.github/workflows/check_pr.yaml

```diff
@@ -0,0 +1,36 @@
+name: CI Check
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    name: Build and Test
+    needs: check-fmt
+    steps:
+      - uses: actions/checkout@v4
+      - uses: moonrepo/setup-rust@v1
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.5
+      - name: Configure sccache env var and set build profile to ephemeral build
+        run: |
+          echo "RUSTC_WRAPPER=sccache" >> $GITHUB_ENV
+          echo "SCCACHE_GHA_ENABLED=true" >> $GITHUB_ENV
+      - run: cargo build --all
+      - run: cargo test --all
+
+  check-fmt:
+    runs-on: ubuntu-latest
+    name: Check formatting
+    steps:
+      - uses: actions/checkout@v4
+      - uses: moonrepo/setup-rust@v1
+        with:
+          components: rustfmt
+      - run: cargo fmt --all -- --check
```

Comment:
I don't see a formatting job as useful.
Especially not, if the library itself isn't formatted properly anyways.

Reply:
This largely builds on [this change](https://github.com/vrchatapi/vrchatapi-rust/blob/978963b2bbfa60bc51f664b16d1c7d4613ab3829/generate.sh#L47) in this PR, kinda driveby i know - if need be ill put in a seperate PR.

Reply:
I already split up two things from your pr.
Try to keep pr's to one thing.

Reply:
You might want to remove those changes from your pr though.
Making smaller pr's is generally better.

Reply:
Sure thing! changed [here](https://github.com/vrchatapi/vrchatapi-rust/pull/17/commits/9be568c4fddd3139a0c6ce1a78befa7717ef7d81)